target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-eap1300.dtsi

```diff
@@ -0,0 +1,237 @@
+
+#include "qcom-ipq4019.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/soc/qcom,tcsr.h>
+
+/ {
+	soc {
+		rng@22000 {
+			status = "okay";
+		};
+
+		mdio@90000 {
+			status = "okay";
+		};
+
+		tcsr@1949000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1949000 0x100>;
+			qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+		};
+
+		ess_tcsr@1953000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1953000 0x1000>;
+			qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+		};
+
+		tcsr@1957000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1957000 0x100>;
+			qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+		};
+
+		crypto@8e3a000 {
+			status = "okay";
+		};
+
+		watchdog@b017000 {
+			status = "okay";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 18 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	aliases {
+		led-boot = &power;
+		led-failsafe = &power;
+		led-running = &power;
+		led-upgrade = &power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power: orange {
+			label = "orange:power";
+			gpios = <&tlmm 58 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "blue:lan";
+			gpios = <&tlmm 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		mesh {
+			label = "blue:mesh";
+			gpios = <&tlmm 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan2g {
+			label = "blue:wlan2g";
+			gpios = <&tlmm 3 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan5g {
+			label = "yellow:wlan5g";
+			gpios = <&tlmm 0 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&tlmm {
+	serial_pins: serial_pinmux {
+		mux {
+			pins = "gpio60", "gpio61";
+			function = "blsp_uart0";
+			bias-disable;
+		};
+	};
+
+	spi_0_pins: spi_0_pinmux {
+		pin {
+			function = "blsp_spi0";
+			pins = "gpio54", "gpio55", "gpio56", "gpio57";
+			drive-strength = <12>;
+			bias-disable;
+		};
+		pin_cs {
+			function = "gpio";
+			pins = "gpio54";
+			drive-strength = <2>;
+			bias-disable;
+			output-high;
+		};
+	};
+};
+
+&blsp_dma {
+	status = "okay";
+};
+
+&blsp1_spi1 {
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	cs-gpios = <&tlmm 54 GPIO_ACTIVE_HIGH>;
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <24000000>;
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "0:SBL1";
+				reg = <0x00000000 0x00040000>;
+				read-only;
+			};
+			partition1@40000 {
+				label = "0:MIBIB";
+				reg = <0x00040000 0x00020000>;
+				read-only;
+			};
+			partition2@60000 {
+				label = "0:QSEE";
+				reg = <0x00060000 0x00060000>;
+				read-only;
+			};
+			partition3@c0000 {
+				label = "0:CDT";
+				reg = <0x000c0000 0x00010000>;
+				read-only;
+			};
+			partition4@d0000 {
+				label = "0:DDRPARAMS";
+				reg = <0x000d0000 0x00010000>;
+				read-only;
+			};
+			partition5@e0000 {
+				compatible = "u-boot,env";
+				label = "0:APPSBLENV";
+				reg = <0x000e0000 0x00010000>;
+
```

Comment:
my guess is it's necessary to remove `read-only;` here to implement it like this?:
https://github.com/openwrt/openwrt/commit/952c738c0fe360ae074fbf6a87d03883157bc2dd

Or could it be left read-only?

Reply:
I'm not sure what you're saying - I did remove read-only so that the u-boot environment can be modified once the system has booted.

Reply:
Is it even possible to parse this partition for macaddresses without
`CONFIG_NVMEM_U_BOOT_ENV=y`
in https://github.com/openwrt/openwrt/blob/master/target/linux/ipq40xx/config-5.15 ?

Reply:
> Is it even possible to parse this partition for macaddresses like this without
CONFIG_NVMEM_U_BOOT_ENV=y

Good catch. I missed that because I'd patched the caldata to include the correct MAC addresses. I will update the PR as soon as I confirm this builds.

Regarding the LAN MAC address, that is being set just fine, but I do not know how.

Reply:
So, to be clear, the LAN MAC address is only used in the device tree to generate the wifi mac addresses. It seems better to just set the LAN MAC address in the device tree, but I don't know how.

Reply:
02_network seems to be used to set the label mac address in uci.

Reply:
Try whether this works :) (It's just an educated guess (I don't know for certain), so if this leads to a bootloop or whatever, then I'm sorry)
https://github.com/openwrt/openwrt/blob/c13263ca07a8d0d7b7eeb02261723f409550c9dd/target/linux/ramips/dts/mt7621_dlink_dir-1935-a1.dts#L15-L18

Reply:
in addition set the alias
`label-mac-device = &gmac0;`

Reply:
It needs to be `mac-address-ascii` and reference `macaddr_lan` under `gmac` but that does seem to work. I'm not going to update this PR with that for now, though.

Reply:
@Djfe Can you please mark this as resolved? There are many other devices being configured by `02_network`. Since I don't understand the implications of moving that to DTS, I would prefer that be a separate PR.